pyomo/solvers/plugins/solvers/GUROBI_RUN.py

```diff
@@ -57,8 +57,14 @@ def gurobi_run(model_file, warmstart_file, soln_file, mipgap, options, suffixes)
             print("***The GUROBI solver plugin cannot extract solution suffix="+suffix)
             return
 
+    # Check for ISV Key License
+    if 'ISV_Key' in options:
+        isv_key_env = Env.OtherEnv(*options['ISV_Key'])
```

Comment:
@DFuller134 Does this not work with
```
isv_key_env = Env(options['ISV_Key'])
```
? I am trying to figure out if there is a more general way to handle all Gurobi environment specifications that also meets your needs. However, I cannot find documentation on Env.OtherEnv. All I see are Env, Env.ClientEnv, and Env.CloudEnv. Can you point me to some docs for Env.OtherEnv?

Reply:
@DFuller134 There is no need to contact Gurobi further. I am thinking of something like the following:
```
env = None
if 'env_type' in options:
    env_type = options['env_type']
    if env_type == 'Env':
        env_class = Env
    elif env_type == 'CloudEnv':
        env_class = Env.CloudEnv
    elif env_type == 'ClientEnv':
        env_class = Env.ClientEnv
    else:
        env_class = Env.OtherEnv
    env = env_class(*(options['env_options']))
```
I think this will support the ISV Key. Thoughts?

Reply:
I like that idea @michaelbynum. Would you object to a single key Env and specification of a tuple (env_type, env_options) ? We might also want to have the key prefixed with Gurobi_ as in Gurobi_Env to generalize the identification/filter of non-Gurobi options.